### PR TITLE
Merge quantifier prenex during prerewrite.

### LINKED ISSUE #10303: Proof regressions since 1.0.5
[bench.tar.gz](https://github.com/cvc5/cvc5/files/14045976/bench.tar.gz)

Hello cvc5 team,

On the attached list of VCs, we see a regression between cvc5 1.0.5 and 1.1.0 in the sense that the old version proved them quickly, and the new version doesn't. This is with the option `--prenex-quant=none`.

Removing the option `--prenex-quant=none` recovers a lot of those VCs, but this option is beneficial for us in general so we would like to keep it ... Or possibly are there other options that could help us with quantifier instantiation? Do you have any suggestions on how to improve performance in our case?

### PR DESCRIPTION
**Pull request by ajreynol**
Fixes #10303.

### CODE REVIEW

File: src/theory/quantifiers/quantifiers_rewriter.h [hunk 1]
```diff
@@ -217,6 +217,12 @@ class QuantifiersRewriter : public TheoryRewriter
   static bool isStandard(QAttributes& qa, const Options& opts);
 
  private:
+  /**
+   * Do trivial merging of the prenex of quantified formula q, e.g.
```
Comment (HanielB) [inline src/theory/quantifiers/quantifiers_rewriter.h:221]: clarify you merge while you can.

File: src/theory/quantifiers/quantifiers_rewriter.cpp [hunk 2]
```diff
@@ -300,6 +281,57 @@ RewriteResponse QuantifiersRewriter::postRewrite(TNode in)
...
+    if (body.getNumChildren() == 2 && body[1].getKind() == k)
+    {
+      QAttributes qa;
+      QuantAttributes::computeQuantAttributes(body[1], qa);
+      // should never combine a quantified formula with a pool or
+      // non-standard quantified formula here.
+      // note that we technically should check
+      // doOperation(body[1], COMPUTE_PRENEX, qa) here, although this
+      // is too restrictive, as sometimes nested patterns should just be
+      // applied to the top level.
```
Comment (HanielB) [inline src/theory/quantifiers/quantifiers_rewriter.cpp:311]: This comment could be more clear. I assume you mean that you'd be able to merge more if you first prenexed the body, right? Moreover, the justification for why it is too restrictive is not clear to me.  Please clarify and give an example.

**Review by HanielB — CHANGES_REQUESTED** [review]

File: src/theory/quantifiers/quantifiers_rewriter.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/quantifiers/quantifiers_rewriter.cpp:311]: FYI this code is not new and is just moved in this PR but I can revisit this.

Comment (HanielB) [reply src/theory/quantifiers/quantifiers_rewriter.cpp]: I noticed but I took the opportunity. :)

**Review by HanielB — APPROVED** [review]

**Pull request merged**